src/trust/index.ts

```diff
@@ -259,3 +262,133 @@ export async function getSignedEntityStatement(
     .then(hasStatusOrThrow(200))
     .then((res) => res.text());
 }
+
+/**
+ * Fetch the federation list document from a given endpoint.
+ *
+ * @param federationListEndpoint The URL of the federation list endpoint.
+ * @param appFetch An optional instance of the http client to be used.
+ * @returns The federation list as an array of strings.
+ * @throws {IoWalletError} If the HTTP request fails or the response cannot be parsed.
+ */
+export async function getFederationList(
+  federationListEndpoint: string,
+  {
+    appFetch = fetch,
+  }: {
+    appFetch?: GlobalFetch["fetch"];
+  } = {}
+): Promise<string[]> {
+  return await appFetch(federationListEndpoint, {
+    method: "GET",
+  })
+    .then(hasStatusOrThrow(200))
+    .then((res) => res.text())
+    .then((text) => {
+      try {
+        return JSON.parse(text) as string[];
```

Comment:
Instead of having a `try/catch` within a `then` block, why not using `Zod` with `safeParse`?

```suggestion
    const federationList = await fetch().then(FederationListResponse.safeParse);
    if (!federationList.success) // throw error
```

Reply:
Thanks, @gispada. I am still not used to Zod, and I forgot to add the validation even if it is a simple type.